week3/homework/src/markDone.js

```diff
@@ -0,0 +1,27 @@
+const write = require('./write');
+const read = require('./read');
+
+function markDone(req, res, toDelete) {
+  read().then(parsedData => {
+    parsedData.map((toDo, index) => {
```

Comment:
why did you decide to use map here instead of find like in the other functions?
filter, find, map etc all take a function as argument which should be a _pure_ function https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976
In other words the function should not have side effects (such as `todo.done = true`)

Reply:
A much nicer way to write would be:

```
read().then(parsedData => {
    const todoItem = parsedData.find(todo => todo.id == req.query.id);
    todoItem.done = ! todoItem.done;
   ....
```

Reply:
yes that's true it's even shorter thanks a lot